### ISSUE TITLE
Grouped Bar Chart with More Than Two Levels

### ISSUE THREAD

**Issue by bacalfa** [ioam/holoviews#2878]
```
bokeh=0.13.0=py36_0
holoviews=1.10.0=py36_1
```

This doesn't seem to be supported yet. From [here](http://holoviews.org/reference/elements/bokeh/Bars.html), it says

> Bars support stacking just like the Area element as well as grouping by a second key dimension.

The following doesn't work as intended (or maybe I'm missing something?):

```
%%opts Bars.Grouped [group_index='Group'] Bars.Stacked [stack_index='Group']
from itertools import product
np.random.seed(3)
index, groups, other = ['A', 'B'], ['a', 'b'], ["x", "y"]
keys = product(index, groups, other)
bars = hv.Bars([k+(np.random.rand()*100.,) for k in keys],
               ["Other", 'Index', 'Group'], 'Count')
bars.relabel(group='Grouped') + bars.relabel(group='Stacked')
```

![image](https://user-images.githubusercontent.com/1297074/42757210-259d4620-88cd-11e8-8942-cc86a7f7a328.png)

**Comment by philippjfr** [comment]
Correct, the BarPlot is currently a bit of a mess. It's not consistent between the matplotlib and bokeh backends (see https://github.com/ioam/holoviews/issues/1407 and https://github.com/ioam/holoviews/issues/1047) and the group_index/stack_index options don't make much sense since you can either group bars or stack them but not both (see https://github.com/ioam/holoviews/issues/2828).

**Comment by HWiese1980** [comment]
Has something changed here? Ticket is almost 2 years old.

**Comment by philippjfr** [comment]
Some things have changed in that bar plot was cleaned up and supports toggling between stacked and unstacked for two levels but more than two levels are still not supported.

**Comment by madeline-scyphers** [comment]
@philippjfr if I wanted to put in a PR to fix this a get the BarPlot to be able to group and stack at the same time, how much work do you think that would be? I’ve ran into this at least half a dozen times at work trying to use holoviews so I would love for it to work. I have a pretty good amount of experience with lower level bokeh and holoviews so I’m pretty sure I can do it.

is this something you’d be interested in? And is anyone else working on it right now?

**Comment by MarcSkovMadsen** [comment]
@philippjfr , @jlstevens. Do you have time to guide a PR here?
